nimble/host/src/ble_sm_sc.c

```diff
@@ -612,6 +612,14 @@ ble_sm_sc_public_key_rx(uint16_t conn_handle, struct os_mbuf **om,
     }
 
     cmd = (struct ble_sm_public_key *)(*om)->om_data;
+    /* Check if the peer public key is same as our generated public key.
+     * Return fail if the public keys match. */
+    if (ble_sm_sc_keys_generated) {
+        if (memcmp(cmd, ble_sm_sc_pub_key, 64) == 0) {
+            res->enc_cb = 1;
+            res->sm_err = BLE_SM_ERR_AUTHREQ;
```

Comment:
shouldn't be here a return?

Could you please create a unit test for that in nimble/host/test/src/ble_sm_sc_test.c

Reply:
Added return. I will create unit test for the same.